core/templates/domain/blog-admin/blog-admin-backend-api.service.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Backend api service for fetching the admin data;
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+interface UserRolesBackendResponse {
+  [role: string]: string;
+}
+
+interface RoleToActionsBackendResponse {
+  [role: string]: string[];
+}
+
+interface ConfigPropertiesBackendResponse {
+  [property: string]: Object;
```

Comment:
Can we make the type definition for the values more specific (`Object` seems a bit generic)?

Reply:
config properties has values of type: `INT_SCHEMA` and `SET_OF_STRINGS_SCHEMA` (for 2 config properties being used in blog admin page). It can also change ( according to scheme defined in config domain) if we add any more config properties to blog admin page. So used `object`

Reply:
I think Object is not good enough. If it changes, we will change here too. Also, we would want the checks to fail on change rather than just covering it up Object. A lot of reasons that were used to remove the any type also apply to Object.

Reply:
Done.